packages/medusa/src/services/strategy-resolver.ts

```diff
@@ -0,0 +1,38 @@
+import { AbstractBatchJobStrategy, TransactionBaseService } from "../interfaces"
+import { EntityManager } from "typeorm"
+import { MedusaError } from "medusa-core-utils"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  [key: string]: unknown
+}
+
+export class StrategyResolver extends TransactionBaseService<
+  StrategyResolver,
+  InjectedDependencies
+> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  constructor(container: InjectedDependencies) {
+    super(container)
+    this.manager_ = container.manager
+  }
+
+  resolveBatchJobByType<T extends TransactionBaseService<never>>(
+    type: string
+  ): AbstractBatchJobStrategy<T> {
+    let resolved: AbstractBatchJobStrategy<T>
+    try {
+      resolved = this.container[
+        `batchType_${type}`
+      ] as AbstractBatchJobStrategy<T>
```

Comment:
fix: this will not create an error; instead of the try-catch you should be able to just do an `if (resolved)`

Reply:
since it is a proxy and not an object, you can see in awilix implementation that the getter is a function that call the awilix internal resolve method. therefore, accessing the property from the proxy that way will trigger a call to the underlying getter implementation and throw an error when the resolve method can't resolve.